### ISSUE TITLE
Favorites and third menu level

### ISSUE THREAD

**Issue by leofeyer** [contao/contao#5401]
As discussed on the developers meeting, we want to improve the back end navigation.

### Favorites

The user shall be able to add any back end URL to a favorites list:

<img width="376" alt="" src="https://user-images.githubusercontent.com/1192057/196389116-404a2bbf-253d-4305-b761-f3c72827c87f.png">

The favorites shall then be displayed on top of the existing back end navigation:

<img width="210" alt="" src="https://user-images.githubusercontent.com/1192057/196389737-79ac299d-62e0-4135-8c98-8253a1e23a51.png">

The favorites shall be organized in a tree structure, so the favorites menu can be nested, too.

### Third menu level

Where applicable, the third menu level shall show the child elements:

<img width="210" alt="" src="https://user-images.githubusercontent.com/1192057/196384986-f8be4aee-c5bb-4303-9067-5e1e8e63378e.png">

Clicking on "Contao Official Demo" will open the child elements view:

<img width="1230" alt="" src="https://user-images.githubusercontent.com/1192057/196385857-683e18de-5fd6-4727-b286-9a9d492bb21a.png">

Maybe we can add a special handling for themes, so the fourth level items (modules, layouts and image sizes) are also directly accessible in the navigation.

**Comment by asaage** [comment]
I like the idea of Favorites.
But a third menu-level in the sidebar not so much.
That would lead to inconsistencies (why can't you navigate the whole site-/ or file-tree from the Sidebar then?)
I figure that would also increase scrolling. You never know how much calendars, newsarchives etc. one might have and the name of them would (even without any indention) most likely not fit in one line of sidebar-space.

**Comment by contaoacademy** [comment]
Perhaps it would be helpful to add custom links to the favorites. Example: A Link to Matomo-Login-URL.

**Comment by contaoacademy** [comment]
It would also be interesting if the data could be exported. So that I can transfer my favorites to another installation.

**Comment by doishub** [comment]
Are the favorites bound to a user or are they defined globally in the project?
If they are bound to a user (which I would prefer), it should also be possible to assign favorites to users as an administrator.

The submenus in the backend could quickly become confusing, which is why I am not completely convinced of the idea.

**Comment by fritzmg** [comment]
> Are the favorites bound to a user or are they defined globally in the project?

Per user.

**Comment by asaage** [comment]
> transfer my favorites to another installation

@contaoacademy i dont think that is practical because you won't have the same articles, news, themes, files, imageSizes (or whatever it is you can favorize) on a different installation.

**Comment by fkaminski** [comment]
Maybe one could also list the favorites links (per user) in the dashboard (in connection with #4503)
